### ISSUE TITLE
Blinds in Plan

### ISSUE THREAD

**Issue by rezzalex** [domoticz/domoticz#5532]
DZ latest version, (build 14928) :

in the "visual" plan, the blinds Icons (Venitian EU in my case) are showing the current status, which is fine.
To Command the Blinds, so change the status (if it is currently open we may want to close it), we currently need to do "a mouse over" first, so that the command button for the other status (close in that example) can be clicked.
![image](https://user-images.githubusercontent.com/35136439/211597996-76c96795-ba1e-4d1a-a903-943e5ca2b468.png)

![image](https://user-images.githubusercontent.com/35136439/211597945-2dbff312-a399-4e05-8cbe-e683c3484005.png)

is it normal behavior?

Would it be possible to have it actionable with only 1 click, without displaying the 2 blind icons first ?

**Comment by waltervl** [comment]
How would you do that 1 click with a blind with position?

**Comment by rezzalex** [comment]
Maybe have a special unique behavior for the plan : when clicking on the current status, "open" for instance, the blind goes down and change status to "Close", except if click on "STOP" (which is currently not here in the plan view).

**Comment by gizmocuz** [comment]
One option would be to add two icons (open/close)

**Comment by jcmylou** [comment]
Hi.

First, I want to thank you for all the work you've done on this extraordinary solution that is Domoticz !

I try to do the same than rezzalex, and it seems that now it "almost" works (version 2024.7) so I wake up this post.

In the visual plan :
- when the blind is close, a click on the blind icon open it and change the state to "Open" => work well
- when the blind is open, a click on the blind icon launch a "switch to OPEN" command instead of CLOSE => have to do a mouse over and select the close option.

Not a real problem if you use it from a laptop but if you want to use it from a phone or a tablet on the wall, it's more complicated (as the "mouse over" is not possible).

The only solution I've found actually is to create a virtual switch which relay a command open/close to the blind switch (15 virtual switchs and LUA script for the whole house) but it's more a workaround than a real solution. Especially since it works with the close command.

Have you a solution for this ? Is it an issue that it works for close but not for open commands ?

Thanks a lot.

**Comment by gizmocuz** [comment]
@jcmylou , could you try beta 16450? The icon image/action should not correctly match the state

**Comment by rezzalex** [comment]
it works, thanks
the cherry on the cake would be to have the stop button displayed somehow

**Comment by jcmylou** [comment]
Hi.

If I can, I prefer not trying a Beta with my main installation :( 
(A lot of configuration to do if the upgrade fail...)

I think I will wait to the next stable version.

Thans to you for your reactivity.

**Comment by waltervl** [comment]
For the stop button see https://forum.domoticz.com/viewtopic.php?p=322542#p322542 for a possible solution

**Comment by gizmocuz** [comment]
> Hi.
> 
> If I can, I prefer not trying a Beta with my main installation :( (A lot of configuration to do if the upgrade fail...)
> 
> I think I will wait to the next stable version.
> 
> Thans to you for your reactivity.

When you upgrade there is a backup made in the backup folder that lets you revert the update.
Another option is to make a complete backup yourself first

The beta will be renamed to release soon

**Comment by rezzalex** [comment]
@gizmocuz , 
I have tested the suggestion in the forum, it is almost perfect : 
we have the stop button available in case of mouseover, clicking "stop" is functionally working but then the blind image is wrong, it shows "blind closed" in my example (it was open and I clicked "close" then "stop").
![image](https://github.com/user-attachments/assets/7c71ec01-4f4d-4fcf-aea4-d52e36abc013)

From my point of view, when "Stop" is clicked, the blind should always be considered open.
(as I was saying in my comments from last 10.01.2023).


Would it be possible to implement the change suggested in the forum + the fix on the status "stop" just explained?

**Comment by gizmocuz** [comment]
@rezzalex , could you test beta 16452?

**Comment by rezzalex** [comment]
Tested with 16452, and not working as expected,

I have re manually changed the domoticzdevice.js file to include the stop button.
when clicking on "stop" button, the blind status is the opposite as it should be: closed instead of open

**Comment by jcmylou** [comment]
Hi,

Always for the same need (but without wanting to abuse) 
would it be possible to add a "**_stop action_**" in addition to "**_open action_**" and "**_close action_**" for the blind+stop switch ?

Like here, in this screenshot :
![image](https://github.com/user-attachments/assets/9d7dede2-7e53-4baf-92f6-6e7f15e6f16d)

Actually, I have to write a lua script which call a bash especially for this case, and it doesn't work perfectly because of the state of the device which is not update in real time but after the end of execution of the previous script.

I have no problem when using the "actions" you've set up.

Thanks

**Comment by gizmocuz** [comment]
When you have a blinds+stop, you should see the stop button with the latest beta version (when you hover over the button)

**Comment by rezzalex** [comment]
> When you have a blinds+stop, you should see the stop button with the latest beta version (when you hover over the button)

No, tried private navigation after switching to latest Beta and the stop button was not there.
As I was saying; I needed to change manually the domoticzdevice.js file to include the stop button.

**Comment by gizmocuz** [comment]
You do not need to change the file by yourself, that is why I changed it for you

See this commit

https://github.com/domoticz/domoticz/commit/4ecf24147cce698f634fea9287efb66c0621cc6a

You do not see the 'stop button until you get the popup

**Comment by rezzalex** [comment]
Mine are "Venitian blinds EU" and I have a stop button, that is why I could not see the stop button

**Comment by gizmocuz** [comment]
Ahhh, that explains.... I will modify the code to support these as well today

**Comment by gizmocuz** [comment]
@rezzalex , could you try beta 16453?

**Comment by rezzalex** [comment]
@gizmocuz , thanks a lot, This is perfect.

when we have the status "Stopped", we don't necessarily want to give the same order again, but when clicked; the text in overlay "Commute to position _open_" + the updated status allows it to be clear.

**Comment by gizmocuz** [comment]
Great it is working. If you think some part could be optimized, please do so by making a PR
